File: GroupAnagrams/GroupAnagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        dic = {}
        for s in strs:
            dic[tuple(sorted(s))] = dic.get(tuple(sorted(s)), []) + [s]
        return [sorted(item) for item in dic.values()]

File: GroupAnagrams/GroupAnagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> help;
        string tmp;
        for(auto s : strs) {
        	tmp = s;
        	sort(tmp.begin(), tmp.end());
        	help[tmp].push_back(s);
        }
        vector<vector<string>> result;
        for(auto h : help) {
        	vector<string> subAnagrams(h.second.begin(), h.second.end());
        	result.push_back(subAnagrams);
        }
        return result;
    }
};